packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts

```diff
@@ -347,13 +349,21 @@ export async function produceBlockBody<T extends BlockType>(
               throw Error(`Missing blobsBundle response from getPayload at fork=${fork}`);
             }
 
-            if (this.opts.sanityCheckExecutionEngineBlobs) {
-              validateBlobsAndKzgCommitments(executionPayload, blobsBundle);
+            let contents: deneb.Contents | fulu.Contents & {cells?: fulu.Cell[][]} = {
+              kzgProofs: blobsBundle.proofs,
+              blobs: blobsBundle.blobs
+            };
+
+            if (ForkSeq[fork] >= ForkSeq.fulu) {
+              const cells = blobsBundle.blobs.map((blob) => ckzg.computeCells(blob));
+              contents = {...contents, cells};
+              validateBlobsAndKzgCommitments(fork as ForkBlobs, executionPayload, blobsBundle, cells);
```

Comment:
```suggestion
             if (this.opts.sanityCheckExecutionEngineBlobs){
               validateBlobsAndKzgCommitments(fork as ForkBlobs, executionPayload, blobsBundle, cells);
             }
```

Reply:
Updated to remove separate call for `validateBlobsAndKzgCommitments`. Now there is only one call, conditional on `opts.sanityCheckExecutionEngineBlobs`